spec/syntax.md

```diff
@@ -481,19 +567,32 @@ text-char = %x0-5B         ; omit \
 
 ### Literals
 
-**_Literal_** is used for matching variants and providing input to _expressions_.
-
-**_Quoted_** literals may include content with any Unicode code point,
+A **_literal_** is a character sequence that appears outside
+of _text_ in various parts of a _message_.
+A _literal_ can appear in a _declaration_, as a _key_ value,
+as an _operand_, or in the value of an _option_.
+A _literal_ MAY include any Unicode code point
 except for surrogate code points U+D800 through U+DFFF.
-The characters `\` and `|` MUST be escaped as `\\` and `\|`.
 
-**_Unquoted_** literals have a much more restricted range that
+All code points are preserved.
+
+A **_quoted_** literal begins and ends with U+005E VERTICAL BAR `|`.
+The characters `\` and `|` within a _quoted_ literal MUST be 
+escaped as `\\` and `\|`.
+
+An **_unquoted_** literal is a _literal_ that does not require the `|`
+quotes around it to be distinct from the rest of the _message_ syntax.
```

Comment:
I think this language makes sense to someone writing a parser, but not to a user necessarily. Perhaps something like "a _literal_ that does not include any of the following characters..." and then refer to where the list of forbidden characters is defined?

Reply:
I tend to agree, but... an unquoted literal is not just a literal that doesn't include specific characters. What marks it as unquoted is that it is missing its quotes! That is, `|42|` is a _quoted_ literal that doesn't need to be quoted.